Support for pre-built indices

nf opened the pull request:
So that you can deploy goimports somewhere without a go tree (eg, App Engine).

adg committed:
make loadExports function override-able

bradfitz commented:
Write some newlines in mkindex.go and squash this pull request into two commits, not 3 or 4.

adg committed:
add mkindex.go

This tool builds a pkgindex.go file that makes it possible to construct
an entirely stand-alone goimports package (usable on App Engine or in
other contexts where GOROOT is not available).

nf commented:
Done

dmitshur commented:
Shouldn't this be noted in README.md? It took me way longer than it should've to figure out what `mkindex.go` does. Getting to this (recent) PR was helpful, but I'd imagine it'll be harder to find as time passes.
